### PR TITLE
Fixed #27175 -- Deprecated silencing exceptions from the {% include %} template tag.

### PR DESCRIPTION
**Pull request by jdufresne**
<https://code.djangoproject.com/ticket/27175>

<https://groups.google.com/forum/#!topic/django-developers/hscW1VboJxY>

### CODE REVIEW

File: django/template/loader_tags.py [hunk 1]
```diff
@@ -208,10 +210,16 @@ def render(self, context):
                 return template.render(context.new(values))
             with context.push(**values):
                 return template.render(context)
-        except Exception:
+        except Exception as e:
             if context.template.engine.debug:
                 raise
             template_name = getattr(context, 'template_name', None) or 'unknown'
+            warnings.warn(
+                "%s raised while rendering {%% include %%} for template '%s'. "
```
Comment (timgraham) [inline django/template/loader_tags.py]: Wording suggestion: "Rendering {% include (template_name) %} raised (error)."

File: django/template/loader_tags.py [hunk 2]
```diff
@@ -208,10 +210,16 @@ def render(self, context):
...
+                "In Django 2.1, this exception will be raised." %
```
Comment (timgraham) [inline django/template/loader_tags.py]: rather than silenced and rendered as an empty string.

File: docs/releases/1.11.txt [hunk 3]
```diff
@@ -514,3 +514,7 @@ Miscellaneous
 * For the ``PyLibMCCache`` cache backend, passing ``pylibmc`` behavior settings
   as top-level attributes of ``OPTIONS`` is deprecated. Set them under a
   ``behaviors`` key within ``OPTIONS`` instead.
+
+* Silencing exceptions raised while rendering the
+  :ttag:`{% include %} <include>` template tag is deprecated. In Django 2.1,
```
Comment (timgraham) [inline docs/releases/1.11.txt]: as the behavior is often more confusing than helpful.

File: docs/internals/deprecation.txt [hunk 4]
```diff
@@ -30,6 +30,9 @@ details on these changes.
 * ``django.core.cache.backends.memcached.PyLibMCCache`` will no longer support
   passing ``pylibmc`` behavior settings as top-level attributes of ``OPTIONS``.
 
+* Silencing exceptions raised while rendering the ``{% include %}`` template
```
Comment (timgraham) [inline docs/internals/deprecation.txt]: Silencing of exceptions

**Comment by timgraham** [comment]
Could you add a test for the deprecation message (even adding it to an existing test is fine) as described in https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#deprecating-a-feature?

**Comment by jdufresne** [comment]
Thanks for the review. I have incorporated your feedback.

File: django/template/loader_tags.py [hunk 5]
```diff
@@ -208,10 +210,16 @@ def render(self, context):
                 return template.render(context.new(values))
             with context.push(**values):
                 return template.render(context)
-        except Exception:
+        except Exception as e:
             if context.template.engine.debug:
                 raise
             template_name = getattr(context, 'template_name', None) or 'unknown'
+            warnings.warn(
+                "Rendering {%% include %s %%} raised %s. In Django 2.1, this "
```
Comment (timgraham) [inline django/template/loader_tags.py]: The template name should have quotes around it.

**Pull request merged**